WordPress/Classes/Utility/Blogging Reminders/BloggingRemindersScheduler.swift

```diff
@@ -276,17 +277,20 @@ class BloggingRemindersScheduler {
             return
         }
 
+        let blogObjectID = blog.objectID
         pushNotificationAuthorizer.requestAuthorization { [weak self] allowed in
-            guard let self = self else {
-                return
-            }
-
-            guard allowed else {
-                completion(.failure(Error.needsPermissionForPushNotifications))
-                return
+            DispatchQueue.main.async {
+                guard let self else { return }
+                guard allowed else {
+                    completion(.failure(Error.needsPermissionForPushNotifications))
+                    return
+                }
+                guard let blog = try? self.coreDataStack.mainContext.existingObject(with: blogObjectID) as? Blog else {
+                    completion(.failure(Error.unknown)) // This should never happen
```

Comment:
IMO, you can just pass the `existingObject(with:)` error to the completion block, instead of creating an "unknown" error here. And, when there is no Core Data error, we can just force cast the fetched `NSManagedObject` instance to Blog—this step should never crash and it's a programming error if it does.

```swift
do {
    let blog = try self.coreDataStack.mainContext.existingObject(with: blogObjectID) as! Blog
    pushAuthorizationReceived(blog: blog, ...)
} catch {
    completion(error)
}
```

Reply:
Agree, it's best not to introduce a new error case for a scenario we don't ever expect to happen. Updated.